### PR TITLE
[IMP] purchase: Make purchase order editable.

### PR DESCRIPTION
**Pull request by dpa-odoo**
Description of the issue/feature this PR addresses:
- Make purchase order editable untill it become locked.

Current behavior before PR:
- Currently purchase order is not editable after draft and sent state.

Desired behavior after PR is merged:
- Now purchase order is editable untill it become locked.
## 

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

Task : https://www.odoo.com/web#id=26666&view_type=form&model=project.task&action=327&menu_id=4720
Pad : https://pad.odoo.com/p/r.c7b93dfc8a76d7f97966167135daf533

### CODE REVIEW

File: addons/purchase/purchase.py [hunk 1]
```diff
@@ -319,15 +320,23 @@ def button_draft(self):
     def button_confirm(self):
         for order in self:
             order._add_supplier_to_product()
-            # Deal with double validation process
+            order._check_double_validation()
+            self.is_editable = False
```
Comment (mdi-odoo) [inline addons/purchase/purchase.py]: May be order.is_editable = False .

File: addons/purchase/purchase.py [hunk 2]
```diff
@@ -319,15 +320,23 @@ def button_draft(self):
...
+        if self.env['ir.values'].get_default('purchase.config.settings', 'auto_done_setting'):
+            self.is_editable = True
```
Comment (mdi-odoo) [inline addons/purchase/purchase.py]: May be order.is_editable = True.

File: addons/purchase/purchase.py [hunk 3]
```diff
@@ -319,15 +320,23 @@ def button_draft(self):
...
+            self._check_double_validation()
```
Comment (mdi-odoo) [inline addons/purchase/purchase.py]: Same here .

File: addons/purchase/purchase.py [hunk 4]
```diff
@@ -319,15 +320,23 @@ def button_draft(self):
     def button_confirm(self):
         for order in self:
             order._add_supplier_to_product()
-            # Deal with double validation process
+            order._check_double_validation()
+            order.is_editable = False
+        if self.env['ir.values'].get_default('purchase.config.settings', 'auto_done_setting'):
+            order.is_editable = True
+            order._check_double_validation()
```
Comment (mdi-odoo) [inline addons/purchase/purchase.py]: Why you need to call **_check_double_validation** method twice ?

File: addons/purchase/purchase.py [hunk 5]
```diff
@@ -319,15 +320,23 @@ def button_draft(self):
     def button_confirm(self):
         for order in self:
             order._add_supplier_to_product()
-            # Deal with double validation process
+            order._check_double_validation()
+            order.is_editable = False
+        if self.env['ir.values'].get_default('purchase.config.settings', 'auto_done_setting'):
```
Comment (mdi-odoo) [inline addons/purchase/purchase.py]: If **Allow to edit purchase orders** is **True** ?

File: addons/purchase/purchase.py [hunk 6]
```diff
@@ -321,13 +322,16 @@ def button_confirm(self):
...
-                    or order.user_has_groups('purchase.group_purchase_manager'):
-                order.button_approve()
+                        or (order.company_id.po_double_validation == 'two_step'\
+                            and order.amount_untaxed < self.env.user.company_id.currency_id.compute(order.company_id.po_double_validation_amount, order.currency_id))\
+                        or order.user_has_groups('purchase.group_purchase_manager'):
+                    order.button_approve()
             else:
                 order.write({'state': 'to approve'})
-        return {}
+                order.is_editable = False
```
Comment (mdi-odoo) [inline addons/purchase/purchase.py]: May be single **write** statement is enough .

File: addons/purchase/purchase.py [hunk 7]
```diff
@@ -322,12 +323,14 @@ def button_confirm(self):
...
             if order.company_id.po_double_validation == 'one_step'\
                     or (order.company_id.po_double_validation == 'two_step'\
-                        and order.amount_total < self.env.user.company_id.currency_id.compute(order.company_id.po_double_validation_amount, order.currency_id))\
+                        and order.amount_untaxed < self.env.user.company_id.currency_id.compute(order.company_id.po_double_validation_amount, order.currency_id))\
                     or order.user_has_groups('purchase.group_purchase_manager'):
                 order.button_approve()
             else:
-                order.write({'state': 'to approve'})
-        return {}
+                order.write({'state': 'to approve', 'is_editable': False})
```
Comment (mdi-odoo) [inline addons/purchase/purchase.py]: Why **is_editable** is **False** ?

**Comment by hmo-odoo** [comment]
merged at https://github.com/odoo/odoo/commit/9d4efc81a6d7563c50d099c426ec5cba43cf11a9

**Pull request closed without merging**